index.html

```diff
@@ -3,449 +3,295 @@
-              size</dfn></a>
-            </li>
-            <li>
-              <a data-cite="HTML#valid-non-negative-integer"><dfn>valid
-              non-negative integer</dfn></a>
+        <li>If |input| has a "type" property that is a [=string=] of length
+        greater than zero:
+          <ol>
+            <li>If |input|'s "type" property value is not a [=valid MIME type
+            string=], return failure.
             </li>
-            <li>
-              <a data-cite="HTML#concept-link-type-sniffing"><dfn>determining
-              the type of the resource</dfn></a>
+            <li>Set |image|'s [=image resource/type=] to the value of |input|'s
+            "type" property.
             </li>
-          </ul>
+          </ol>
         </li>
-        <li>[[MIMESNIFF]] defines the following terms:
-          <ul>
-            <li>
-              <a data-cite="MIMESNIFF#mime-type"><dfn>MIME type</dfn></a>
-            </li>
-            <li>
-              <a data-cite="MIMESNIFF#valid-mime-type-string"><dfn>valid MIME
-              type</dfn></a>
-            </li>
-          </ul>
+        <li>Return |image|.
         </li>
-        <li>[[INFRA]] defines the following terms:
-          <ul>
-            <li>
-              <dfn data-cite="INFRA#ascii-lowercase" data-lt="lowercased">ASCII
-              lowercase</dfn>
-            </li>
-            <li>
-              <dfn data-cite="INFRA#string">string</dfn>
-            </li>
-            <li>
-              <dfn data-cite="INFRA#iteration-continue">continue</dfn>
-            </li>
-            <li>
-              <dfn data-cite="INFRA#list">list</dfn>
-            </li>
-            <li>
-              <dfn data-cite="INFRA#list-iterate">for each</dfn> (for list)
-            </li>
-            <li>
-              <dfn data-cite="INFRA#list-contain">contains</dfn>
+      </ol>
+    </section>
+    <section>
+      <h2>
+        Fetching an image resource
+      </h2>
+      <p>
+        The steps for <dfn data-export="">fetching an image resource</dfn> are
+        given by the following algorithm. The algorithm takes an [=image
+        resource=] |image|, and an optional [=Request=] |request|. It returns a
+        [=Response=].
+      </p>
+      <ol>
+        <li>If |request| is `undefined`:
+          <ol>
+            <li>Set |request| to a new [=Request=].
             </li>
-            <li>
-              <dfn data-cite="INFRA#list-is-empty">is empty</dfn>
+            <li>Set [=request/URL=] to |image|'s [=image resource/src=].
             </li>
-            <li>
-              <dfn data-cite="INFRA#list-append" data-lt=
-              "list-append">append</dfn>
+            <li>Set [=request/Destination=] to "image".
             </li>
-            <li>
-              <dfn data-cite="INFRA#set">set</dfn>
+            <li>Set [=request/Mode=] to "no-cors".
             </li>
-          </ul>
+          </ol>
```

Comment:
Feels like the client and credentials mode of this request should be set?

Reply:
Fwiw, see the icon fetching in https://wicg.github.io/background-fetch/#background-fetch-display-algorithm

Reply:
Different specs tend to use different Request parameters to fetch the icon, which is why the algorithm accepts a custom request. For example there is little overlap between how BGF fetches the icon and how [WebAppManifest](https://www.w3.org/TR/appmanifest/#fetching-image-resources) handles it.

Reply:
I wonder if we should provide some guidance there? Or set some defaults if they haven't been set?

Reply:
It currently defaults to the Request object's defaults plus the following:
- src -> set the the processed image resource's src
- destination -> "image"
- mode -> "no cors"

That felt like minimum amount of information needed that doesn't make assumptions on the API's behalf.

Reply:
thanks @rayankans. I defer to @jakearchibald.